src/PlaylistSave.cxx

```diff
@@ -84,20 +84,29 @@ playlist_print_uri(BufferedOutputStream &os, const char *uri)
 }
 
 void
-spl_save_queue(const char *name_utf8, const Queue &queue)
+spl_save_queue(const char *name_utf8, PlaylistSaveMode save_mode, const Queue &queue)
 {
 	const auto path_fs = spl_map_to_fs(name_utf8);
 	assert(!path_fs.IsNull());
 
-	if (FileExists(path_fs))
-		throw PlaylistError(PlaylistResult::LIST_EXISTS,
-				    "Playlist already exists");
+	if (save_mode == PlaylistSaveMode::CREATE) {
+		if (FileExists(path_fs)) {
+			throw PlaylistError(PlaylistResult::LIST_EXISTS, "Playlist already exists");
+		}
+	}
+	else if (!FileExists(path_fs)) {
+		throw PlaylistError(PlaylistResult::NO_SUCH_LIST, "No such playlist");
+	}
 
-	FileOutputStream fos(path_fs);
+	FileOutputStream fos(path_fs,
+						 save_mode == PlaylistSaveMode::APPEND
+										 ? FileOutputStream::Mode::APPEND_EXISTING
+										 : FileOutputStream::Mode::CREATE);
 	BufferedOutputStream bos(fos);
 
-	for (unsigned i = 0; i < queue.GetLength(); i++)
+	for (unsigned i = 0; i < queue.GetLength(); i++) {
 		playlist_print_song(bos, queue.Get(i));
+	}
```

Comment:
Don't change existing code style (in an unrelated commit); this adds unnecessary noise to the diff.